core/src/toga/images.py

```diff
@@ -3,34 +3,40 @@
 import warnings
 from io import BytesIO
 from pathlib import Path
+from typing import TYPE_CHECKING, TypeVar
 from warnings import warn
 
 try:
-    import PIL.Image as PIL_Image
+    import PIL.Image
+
+    PIL_imported = True
 except ImportError:  # pragma: no cover
-    PIL_Image = None
+    PIL_imported = False
 
 import toga
 from toga.platform import get_platform_factory
 
+if TYPE_CHECKING:
+    ImageT = TypeVar("ImageT", bound=toga.Image | PIL.Image.Image)
+
 # Make sure deprecation warnings are shown by default
 warnings.filterwarnings("default", category=DeprecationWarning)
 
 
 class Image:
     def __init__(
         self,
-        src: str | Path | bytes | PIL_Image.Image | None = None,
+        src: str | Path | bytes | PIL.Image.Image | None = None,
         *,
         path=None,  # DEPRECATED
         data=None,  # DEPRECATED
     ):
         """Create a new image.
 
         :param src: The source from which to load the image. Can be a file path
-            (relative or absolute, as a string or :external:class:`pathlib.Path`) or raw
+            (relative or absolute, as a string or :external:any:`pathlib.Path`) or raw
             binary data in any supported image format. Can also accept a
-            :external:class:`PIL.Image.Image` if Pillow is installed.
+            :external:any:`PIL.Image.Image` if Pillow is installed.
```

Comment:
I'm pretty sure the `:external:` is redundant too - `:any:` will resolve the best match, including external project references.